pkgs/servers/matrix-synapse/default.nix

```diff
@@ -44,47 +46,93 @@ buildPythonApplication rec {
     rustc
   ]);
 
-  buildInputs = [ openssl ];
+  buildInputs = [
+    openssl
+  ];
 
-  propagatedBuildInputs = [
-    authlib
+  propagatedBuildInputs = with python3.pkgs; [
+    attrs
     bcrypt
     bleach
     canonicaljson
-    daemonize
+    cryptography
     frozendict
     ijson
     jinja2
     jsonschema
-    lxml
     matrix-common
     msgpack
     netaddr
+    packaging
     phonenumbers
     pillow
     prometheus-client
-    psutil
-    psycopg2
     pyasn1
+    pyasn1-modules
     pydantic
-    pyicu
     pymacaroons
-    pynacl
     pyopenssl
-    pysaml2
     pyyaml
-    requests
-    setuptools
+    service-identity
     signedjson
     sortedcontainers
     treq
     twisted
     typing-extensions
     unpaddedbase64
-  ] ++ lib.optional enableSystemd systemd
-    ++ lib.optionals enableRedis [ hiredis txredisapi ];
+  ]
+  # extras
+  ++ twisted.optional-dependencies.tls;
 
-  nativeCheckInputs = [ mock parameterized openssl ];
+  passthru.optional-dependencies = with python3.pkgs; {
+    matrix-synapse-ldap3 = with plugins; [
+      matrix-synapse-ldap3
+    ];
+    postgres = if isPyPy then [
+      psycopg2cffi
+    ] else [
+      psycopg2
+    ];
+    saml2 = [
+      pysaml2
+    ];
+    oidc = [
+      authlib
+    ];
+    systemd = lib.optionals (lib.meta.availableOn stdenv.hostPlatform systemd) [
+      systemd
+    ];
```

Comment:
I would rather report this as an assert error on the module level

Reply:
This is really an edge case and just makes sure that we provide it only, if supported. I don't think I want to invest much more logic here.

Reply:
How about extraFeatures?

Reply:
Good point. Can't we just turn this optionals into an assertion (I'm fine with `throwIf` for instance).

Reply:
Would it be enough to just remove the conditional on the list? Then accessing it on an unsupported platform would throw by itself.

Reply:
good point, fine by me!